src/Program.cs

```diff
@@ -9,95 +9,149 @@ public class Program {
             // input is what we get from the user
             string input = "";
 
-            // set up a new board and display it
-            Console.WriteLine("Starting game...");
-            MoveState[] board = new MoveState[9]; // 3x3 board = 9 length
-            ResetBoard(board);
-            DisplayBoard(board);
-
-            // actual game
-            // runs until "quit" is provided
-            while (true) {
+            // determining difficulty
+            while (input != "quit") {
+                // difficulty system
                 // retrieving the input
                 try {
-                    input = RetrieveInput();
+                    input = RetrieveInput("Enter \"quit\" to end the game. Enter the difficulty: \"Easy\", \"Medium\", \"Hard\", or \"Impossible\"").ToLower();
                 } catch (IOException e) {
                     Console.WriteLine(e.Message);
                     continue;
                 }
 
-                // ends the game if "quit" is provided
-                if (input.ToLower() == "quit") break;
+                // exiting if needed
+                if (input == "quit") break;
 
-                // ignore 'empty' input
+                // ignoring 'empty' input
                 if (input == "") continue;
 
-                // parsing the input to ensure it is a valid number in the given range
-                int num = 0;
-                if (!int.TryParse(input, out num)) {
-                    Console.WriteLine("Invalid input provided. Not a number or \"quit\"");
-                    continue;
-                }
-                if (num < 1 || num > 9) {
-                    Console.WriteLine("Number outwith range (1-9)");
-                    continue;
+                // getting the relevant enum
+                AIChance difficulty = AIChance.Impossible;
+                bool reset = false;
+                switch (input) {
+                    case "easy":
+                        difficulty = AIChance.Easy;
+                        break;
+                    case "medium":
+                        difficulty = AIChance.Medium;
+                        break;
+                    case "hard":
+                        difficulty = AIChance.Hard;
+                        break;
+                    case "impossible":
+                        difficulty = AIChance.Impossible;
+                        break;
+                    default:
+                        Console.WriteLine("Invalid input provided. Please enter \"easy\", \"medium\", \"hard\", or \"impossible\"");
+                        reset = true;
```

Comment:
you can remove the bool and replace this part with `continue`:

```csharp
default:
    Console.WriteLine("Invalid input provided. Please enter \"easy\", \"medium\", \"hard\", or \"impossible\"");
    continue;
```

Reply:
good point

Reply:
![pee blue](https://github.com/VictorVictini/TicTacToe/assets/100409468/1c139879-a632-4ca3-89f4-830c7dca189b)

Reply:
added the change